File: Examples2018/UIToggleButton/app/src/main/java/course/examples/ui/togglebutton/ToggleButtonActivity.java
package course.examples.ui.togglebutton;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.FrameLayout;
import android.widget.Switch;
import android.widget.ToggleButton;

public class ToggleButtonActivity extends Activity {
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.main);

        // Get references to a background containers
        final FrameLayout top = findViewById(R.id.top_frame);
        final FrameLayout bottom = findViewById(R.id.bottom_frame);


		// Get a reference to the ToggleButton
        final ToggleButton toggleButton = findViewById(R.id.togglebutton);

        // Set an setOnCheckedChangeListener on the ToggleButton
        setListener(toggleButton, top);

        // Get a reference to the Switch
        final Switch switcher = findViewById(R.id.switcher);

        // Set an OnCheckedChangeListener on the Switch
        setListener(switcher, bottom);

    }

    private void setListener(CompoundButton button, final View background) {

        button.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
                // Toggle the Background color between a light and dark color
                if (isChecked) {
                    background.setBackgroundColor(getResources().getColor(R.color.primary_light));
                } else {
                    background.setBackgroundColor(Color.TRANSPARENT);
                }
			}
		});
	}
}

File: ExamplesKotlin/UIToggleButton/app/src/main/java/course/examples/ui/togglebutton/ToggleButtonActivity.kt
package course.examples.ui.togglebutton

import android.app.Activity
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.CompoundButton
import android.widget.FrameLayout
import android.widget.Switch
import android.widget.ToggleButton

class ToggleButtonActivity : Activity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        // Set an setOnCheckedChangeListener on the ToggleButton
        setListener(
            findViewById<ToggleButton>(R.id.togglebutton),
            findViewById<FrameLayout>(R.id.top_frame)
        )

        // Set an OnCheckedChangeListener on the Switch
        setListener(
            findViewById<Switch>(R.id.switcher),
            findViewById<FrameLayout>(R.id.bottom_frame)
        )

    }

    private fun setListener(button: CompoundButton, background: View) {

        button.setOnCheckedChangeListener { _, isChecked ->
            // Toggle the Background color between a light and dark color
            if (isChecked) {
                background.setBackgroundColor(resources.getColor(R.color.primary_light, null))
            } else {
                background.setBackgroundColor(Color.TRANSPARENT)
            }
        }
    }
}